apps/studio/pages/api/ai/sql/tools.ts

```diff
@@ -387,5 +387,179 @@ export const getTools = ({
         }
       },
     }),
+    getEdgeFunctionKnowledge: tool({
+      description: 'Get knowledge about how to write edge functions for Supabase',
+      parameters: z.object({}),
+      execute: async ({}) => {
+        console.log('schemas', 'edge function called...')
+        return stripIndent`
+        # Writing Supabase Edge Functions
+
+        You're an expert in writing TypeScript and Deno JavaScript runtime. Generate **high-quality Supabase Edge Functions** that adhere to the following best practices:
+
+        ## Guidelines
+
+        1. Try to use Web APIs and Deno's core APIs instead of external dependencies (eg: use fetch instead of Axios, use WebSockets API instead of node-ws)
+        2. If you are reusing utility methods between Edge Functions, add them to \`supabase/functions/_shared\` and import using a relative path. Do NOT have cross dependencies between Edge Functions.
+        3. Do NOT use bare specifiers when importing dependecnies. If you need to use an external dependency, make sure it's prefixed with either \`npm:\` or \`jsr:\`. For example, \`@supabase/supabase-js\` should be written as \`npm:@supabase/supabase-js\`.
+        4. For external imports, always define a version. For example, \`npm:@express\` should be written as \`npm:express@4.18.2\`.
+        5. For external dependencies, importing via \`npm:\` and \`jsr:\` is preferred. Minimize the use of imports from @\`deno.land/x\` , \`esm.sh\` and @\`unpkg.com\` . If you have a package from one of those CDNs, you can replace the CDN hostname with \`npm:\` specifier.
+        6. You can also use Node built-in APIs. You will need to import them using \`node:\` specifier. For example, to import Node process: \`import process from "node:process"\`. Use Node APIs when you find gaps in Deno APIs.
+        7. Do NOT use \`import { serve } from "https://deno.land/std@0.168.0/http/server.ts"\`. Instead use the built-in \`Deno.serve\`.
+        8. Following environment variables (ie. secrets) are pre-populated in both local and hosted Supabase environments. Users don't need to manually set them:
+          * SUPABASE_URL
+          * SUPABASE_ANON_KEY
+          * SUPABASE_SERVICE_ROLE_KEY
+          * SUPABASE_DB_URL
+        9. To set other environment variables (ie. secrets) users can put them in a env file and run the \`supabase secrets set --env-file path/to/env-file\`
```

Comment:
Do we have any strategies around this via UI? Link them to the secrets settings page? Even better: inline chat component to set secrets?

Reply:
This is copied from the general AI prompt (which may not always have access to Studio UI). But I suggested to @SaxonF in the Studio specific prompt which should suggest to use the Secrets UI over this (can we override this rule in Studio assistant?).